src/main/scala/is/hail/variant/VariantView.scala

```diff
@@ -0,0 +1,56 @@
+package is.hail.variant
+
+import is.hail.annotations._
+import is.hail.expr._
+
+class VariantView(t: TStruct) {
+  private var region: MemoryBuffer = _
+  private var offset: Long = _
+  private val contigIdx: Int = t.fieldIdx("contig")
+  private val startIdx: Int = t.fieldIdx("start")
+  private val refIdx: Int = t.fieldIdx("ref")
+  private val altAllelesIdx: Int = t.fieldIdx("altAlleles")
+  private val tAltAlleles: TArray = t.fieldType(altAllelesIdx).asInstanceOf[TArray]
+  private val tAltAllele: TStruct = tAltAlleles.elementType.asInstanceOf[TStruct]
+  private val altIdx: Int = tAltAllele.fieldIdx("alt")
+
+  assert(t.fieldType(contigIdx).required)
+  assert(t.fieldType(startIdx).required)
+  assert(t.fieldType(refIdx).required)
+  assert(tAltAlleles.required)
+  assert(tAltAlleles.elementType.required)
+  assert(tAltAllele.fieldType(altIdx).required)
```

Comment:
It seems reasonable to hardcode all of this based on `TVariant`'s representation.

Reply:
What do you mean by hard code this? I am using these asserts (which perhaps should be `require`s) to make explicit the properties of the representation on which this class depends.